message_definitions/v1.0/loweheiser.xml

```diff
@@ -0,0 +1,55 @@
+  <include>minimal.xml</include>
+  <enums>
+    <enum name="MAV_CMD">
+      <!-- Loweheiser specific MAV_CMD_* commands -->
+      <entry name="MAV_CMD_LOWEHEISER_SET_STATE" value="10151">
+        <description>Set Loweheiser desired states</description>
+        <param index="1">EFI Index</param>
+        <param index="2">Desired Engine/EFI State (0: Power Off, 1:Running)</param>
+        <param index="3">Desired Governor State (0:manual throttle, 1:Governed throttle)</param>
+        <param index="4">Manual throttle level, 0% - 100%</param>
+        <param index="5">Electronic Start up (0:Off, 1:On)</param>
+        <param index="6">Empty</param>
+        <param index="7">Empty</param>
+      </entry>
+    </enum>
+  </enums>
+  <messages>
+    <message id="10151" name="LOWEHEISER_GOV_EFI">
+      <description>Composite EFI and Governor data from Loweheiser equipment.  This message is created by the EFI unit based on its own data and data received from a governor attached to that EFI unit.</description>
+      <!-- Generator fields -->
+      <field type="float" name="volt_batt" units="volts">Generator Battery voltage.</field>
+      <field type="float" name="curr_batt" units="A">Generator Battery current.</field>
+      <field type="float" name="curr_gen" units="A">Current being produced by generator.</field>
+      <field type="float" name="curr_rot" units="A">Load current being consumed by the UAV (sum of curr_gen and curr_batt)</field>
+      <field type="float" name="fuel_level" units="l">Generator fuel remaining in mL.</field>
+      <field type="float" name="throttle" units="%">Throttle Output.</field>
+      <field type="uint32_t" name="runtime" units="s">Seconds this generator has run since it was rebooted.</field>
+      <field type="int32_t" name="until_maintenance" units="s">Seconds until this generator requires maintenance.  A negative value indicates maintenance is past due.</field>
+      <field type="float" name="rectifier_temp" units="degC">The Temperature of the rectifier.</field>
+      <field type="float" name="generator_temp" units="degC">The temperature of the mechanical motor, fuel cell core or generator.</field>
+      <!-- EFI fields -->
+      <field type="float" name="efi_batt" units="volts"> EFI Supply Voltage.</field>
+      <field type="float" name="efi_rpm" units="rpm">Motor RPM.</field>
+      <field type="float" name="efi_pw" units="ms">Injector pulse-width in miliseconds.</field>
+      <field type="float" name="efi_fuel_flow" units="l/h">Fuel flow rate.</field>
+      <field type="float" name="efi_fuel_consumed" units="l">Fuel consumed.</field>
+      <field type="float" name="efi_baro" units="kPa">Atmospheric pressure.</field>
+      <field type="float" name="efi_mat" units="degC">Manifold Air Temperature.</field>
+      <field type="float" name="efi_clt" units="degC">Cylinder Head Temperature.</field>
+      <field type="float" name="efi_tps" units="%">Throttle Position.</field>
```

Comment:
I really want to rename these fields to be descriptive.....
```suggestion
      <field type="float" name="efi_man_air_temp" units="degC">Manifold Air Temperature.</field>
      <field type="float" name="efi_cyl_head_temp" units="degC">Cylinder Head Temperature.</field>
      <field type="float" name="efi_throttle_pos" units="%">Throttle Position.</field>
```

Reply:
This is what the vendor wanted.  I did argue at the time, but at the end of the day it wasn't worth my time arguing...

Reply:
Hi!
In my opinion for easy working, the initials are better becasuse the re is a description at the defition of the field. the only field i see it could be better is change "efi_clt" to "efi_cht".

But we are open to talk for all changes that you suggest @peterbarker 
Have a nice day!

Reply:
Hi @QUAV ,

Unfortunately, in a lot of log analysis tools those description fields don't show up. 

Plus being descriptive is nice to those looking at or graphing messages in real-time or looking at a mavlink inspector.

Reply:
I agree with you

Reply:
We are agree about that